crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -477,10 +475,7 @@ impl ShaderProcessor {
             Source::Wgsl(source) => source.deref(),
             Source::Glsl(source, _stage) => source.deref(),
             Source::SpirV(source) => {
-                if shader_defs_unique.is_empty() {
-                    return Ok(ProcessedShader::SpirV(source.clone()));
-                }
-                return Err(ProcessShaderError::ShaderFormatDoesNotSupportShaderDefs);
+                return Ok(ProcessedShader::SpirV(source.clone()));
```

Comment:
Would it makes sense to log a message saying something like "loaded a SpirV shader, it will ignore shader defs"?

Reply:
I'd have thought that would be implicit, or something that would belong documentation-side if anything; the processor is either interested in SPIR-V or not, so my initial feeling is that it would end up filling the log with obvious statements in a project that makes extensive use of the format.

However, @superdump suggested an alternate approach on Discord that seems less intrusive; keeping the hard error, but giving `Material` implementors the opportunity to clear out all built-in shader defs at specialization time.

Currently `NO_*_SUPPORT` and `SIXTEEN_BYTE_ALIGNMENT`get injected after the fact and create the hard error situation this PR aims to remedy, so moving them earlier in the pipeline would avoid breaking the processor's existing assumptions while simultaneously allowing `Material` the control it needs for the use-case.

I was planning to open another pull to that effect anyway, so will take a swing at it and see where things go from there.

Reply:
New PR: https://github.com/bevyengine/bevy/pull/7903